demos/exyielding/README.md

```diff
@@ -0,0 +1,25 @@
+# Exyielding
+
+In this demo we want to show that a thread can yield the utilization of the CPU directly to another thread.
+
+To illustrate the mechanism in it, we will explain:
+
+* `yield_to` (in `src/user/thread.c`)
+
+## `yield_to`
+
+When this function is called, the kernel will check if the thread we specified is available to yield to.
+
+If the thread can be yielded to, we will set `next_thread` to the address of this thread and call the `do_scheduler` function.
+
+Since we have decided `next_thread`, the scheduler will do housekeeping tasks and help us to setup the thread we had already chosen.
+
+If the thread we specified is unavailable to yield, `do_scheduler` will not be called and we will go back to the point we called `yield_to()`.
+
+## Walkthrough
+
+At the beginning of this demo, we create two named thread  (`first` and `second`). After the thread (`setup`) ends, the thread (`first`) will get the use of the CPU and yield to the thread `second` specified by its thread ID (`tid`).
```

Comment:
Remove the double space in "named thread  (".

Also you can remove the brackets around setup/first/second. Reads fine without them.